### ISSUE TITLE
`add --from bibtex` should work with file or string

### ISSUE THREAD

**Issue by pinusc** [papis/papis#838]
* papis version (`$ papis --version` or commit number):  8c01969

It should be easy to import citations from a bibtex file (or string!) when running `papis add`.

## Current Behavior
Right now, the only thing that works is 

```
papis add --from bibtex https://someUrlOfABibtexFile.html theArticle.pdf
```

## Desired Behavior
I expect that the following should work:

```
papis add --from bibtex theCitation.bib theArticle.pdf
```

or even:

```
papis add --from bibtex "@article{someRef,
author={theAuthor},
...
}" theArticle.pdf
```

(while it may seem like a strange workflow, copying the bibtex from somewhere else and then pasting in the terminal is quite easy and less hassle than creating a temporary file and then deleting it after).

If I'm not mistaken, the only available workflow to do the above currently is 

```
papis bibtex read theCitation.bib import --all
papis addto -f theArticle.pdf
```

This seems silly; why would I need to remember a different command to do the same thing I can usually do easily from URL (or doi, isbn...)?

## Implementation
It also feels like it should be easy to implement. If the argument supplied to "--from bibtex" starts with a '@', import it as a bibtex string. If it starts with 'http', then do what it currently does (download the URL and import that as a bibtex file). Otherwise, check if the provided string corresponds to a filename and import that.

**Comment by alexfikl** [comment]
Agreed, it would be nice for these to just work™! I'm a bit surprised this doesn't work:
```
papis add --from bibtex theCitation.bib theArticle.pdf
```
but it would also be nice to be able to do
```
papis add --from bibtex $(get_bibtex from_somewhere) theArticle.pdf
```
or just copy paste it in there like you said.

The code for `papis add --from bibtex <URL>` is here
https://github.com/papis/papis/blob/8c0196981242e4b7342ee87fc9088c13df8f80cd/papis/bibtex.py#L303-L345
and it should be pretty easy to teach it to be more lenient about what it accepts.

**Comment by pinusc** [comment]
> Agreed, it would be nice for these to just work™! I'm a bit surprised this doesn't work:
> 
> ```
> papis add --from bibtex theCitation.bib theArticle.pdf
> ```

Thank you for your reply. I started checking out the code thinking it might be a great time to finally submit my first contribution to papis, and...

Turns out _it does work_ and I, an idiot, was actually dealing with a malformed bibtex file / papis command. So at least that's what one would expect.

And that does actually solve most of the issues with my workflow, especially since process substitution exists:

```
papis add --from bibtex <(get_bibtex from_somewhere) theArticle.pdf
```

which actually works. However, I feel that to most people, this is obscure enough bash/zsh knowledge that having papis just handle bibtex strings would be easier.

**Comment by pinusc** [comment]
Well, I am officially an idiot. Everything _already works_ as intended and as I requested, including handling strings.

If you wonder what went wrong that made me think it didn't, here's a post-mortem:

1. First, I had tried importing a bibtex file that missed a reference, so it started with `@incollection{,` which is obviously incorrect. After fixing that, what do you know, `papis add --from bibtex theBibtex.bib` worked!
2. I am a smart-ass and when I want to paste into my terminal I usually use `wl-paste` (or `xclip` in my X11 days). However, I can't bother to type `"$(wl-paste)"`, and usually end up typing `` `wl-paste` `` instead. I was on the _very wrong_ impression that backtick substitution didn't need quotes. `` `wl-paste` `` mangled my input and passed each line as a separate argument (I think); so papis was only trying to parse _the first line_.

In conclusion, this was entirely user error based on incorrect usage of my shell.

In my defense, the error reporting was a bit confusing for the second case. It read:

```
[INFO] importer.bibtex: Reading input file: '@incollection{theReference,'.
[WARNING] importer.bibtex: Empty or invalid BibTeX entry at '@incollection{theReference,'.
```

which really does sound like papis took my string, truncated it at newline, and tried to open a file by that name. Looking at the code, it's probably non-trivial to make that message more explicit. 

Sorry for wasting everyone's time

**Comment by pinusc** [comment]
Thanks for the nudge! I opened a quick pull request adding some examples and improving error message wording
